### PR TITLE
chore: add serverless monitoring

### PR DESCRIPTION
**Pull request by YiniXu9506**
image

### CODE REVIEW

File: ui/packages/tidb-dashboard-for-dbaas/src/apps/Monitoring/metricsQueries.ts [hunk 1]
```diff
@@ -61,7 +62,7 @@ const getMonitoringItems = (
     return 'sum(tikv_store_size_bytes{type="used"}) by (instance)'
   }
 
-  const monitoringItems: MetricsQueryType[] = [
+  const monitoringItems_dedicated: MetricsQueryType[] = [
```
Comment (baurine) [inline ui/packages/tidb-dashboard-for-dbaas/src/apps/Monitoring/metricsQueries.ts:65]: Prefer to use camel naming style, likes: `dedicatedMonitoringItems`.

**Review by shhdgit — APPROVED** [review]

File: ui/packages/tidb-dashboard-for-dbaas/src/apps/Monitoring/metricsQueries.ts [hunk 2]
```diff
@@ -766,7 +767,159 @@ const getMonitoringItems = (
     }
   ]
 
-  return monitoringItems
+  const monitoringItems_serverless: MetricsQueryType[] = [
```
Comment (baurine) [inline ui/packages/tidb-dashboard-for-dbaas/src/apps/Monitoring/metricsQueries.ts:770]: Prefer to use camel naming style, likes: `serverlessMonitoringItems`.

**Review by baurine — APPROVED** [review]

File: ui/packages/tidb-dashboard-for-dbaas/src/apps/Monitoring/context.ts [hunk 3]
```diff
@@ -62,10 +62,12 @@ export const ctx: (globalConfig: IGlobalConfig) => IMonitoringContext = (
...
     getMetricsQueries: (pdVersion: string | undefined) =>
-      getMonitoringItems(pdVersion),
+      getMonitoringItems(pdVersion, globalConfig.clusterInfo.deployType),
     timeRangeSelector: {
       recent_seconds: RECENT_SECONDS,
       customAbsoluteRangePicker: true
-    }
+    },
+    metricsWithoutCategory:
+      globalConfig.clusterInfo.deployType === 'Dedicated' ? false : true
```
Comment (baurine) [inline ui/packages/tidb-dashboard-for-dbaas/src/apps/Monitoring/context.ts:71]: `globalConfig.clusterInfo.deployType === 'Dedicated'` is fine, it returns boolean

File: ui/packages/tidb-dashboard-lib/src/apps/Monitoring/components/Monitoring.tsx [hunk 4]
```diff
@@ -111,56 +111,99 @@ export default function Monitoring() {
...
+                        onClickSeriesLabel={(seriesName) =>
+                          telemetry.clickSeriesLabel(item.title, seriesName)
+                        }
+                      />
+                    </Card>
+                  </Col>
+                ))}
+              </Row>
+            ) : (
+              <>
```
Comment (baurine) [inline ui/packages/tidb-dashboard-lib/src/apps/Monitoring/components/Monitoring.tsx:151]: this element `<>` is redundant.

File: ui/packages/tidb-dashboard-lib/src/apps/Monitoring/components/Monitoring.tsx [hunk 5]
```diff
@@ -111,56 +111,99 @@ export default function Monitoring() {
...
+                                  telemetry.clickSeriesLabel(
+                                    item.title,
+                                    seriesName
+                                  )
+                                }
+                              />
+                            </Card>
+                          </Col>
+                        ))}
+                      </Row>
```
Comment (baurine) [inline ui/packages/tidb-dashboard-lib/src/apps/Monitoring/components/Monitoring.tsx:201]: This part looks duplicated.

**Comment by baurine** [comment]
The latest commit make the PR not concise anymore, need to refine.

**Review by baurine — APPROVED** [review]

File: ui/packages/tidb-dashboard-lib/src/apps/Overview/components/LimitTimeRange.tsx [hunk 6]
```diff
@@ -43,12 +43,22 @@ export const LimitTimeRange: React.FC<LimitTimeRangeProps> = ({
   }, [recent_seconds])
 
   const disabledDate = (current) => {
-    const today = dayjs().startOf('hour')
+    const todayStartWithHour = dayjs().startOf('hour')
+    const todayStartWithDay = dayjs().startOf('day')
+    const todayEndWidthDay = dayjs().endOf('day')
```
Comment (baurine) [inline ui/packages/tidb-dashboard-lib/src/apps/Overview/components/LimitTimeRange.tsx:48]: `todayEndWithDay`

Comment (baurine) [reply ui/packages/tidb-dashboard-lib/src/apps/Overview/components/LimitTimeRange.tsx:48]:
```
    const today = dayjs()
    const todayStartWithHour = today.startOf('hour')
    const todayStartWithDay = today.startOf('day')
    const todayEndWidthDay = today.endOf('day')
```

File: ui/packages/tidb-dashboard-lib/src/apps/Overview/components/LimitTimeRange.tsx [hunk 7]
```diff
@@ -43,12 +43,22 @@ export const LimitTimeRange: React.FC<LimitTimeRangeProps> = ({
...
+      todayStartWithHour.subtract(selectableHours, 'hour') >
+        dayjs(current).startOf('hour') &&
+      todayStartWithDay.subtract(selectableHours / 24, 'day') >
+        dayjs(current).startOf('day')
+
     // Can not select days after today
-    const tooLate = today < dayjs(current).startOf('hour')
+    const tooLate =
+      todayStartWithHour < dayjs(current).startOf('hour') &&
+      todayEndWidthDay < dayjs(current).endOf('day')
```
Comment (baurine) [inline ui/packages/tidb-dashboard-lib/src/apps/Overview/components/LimitTimeRange.tsx:60]:
```
const curDate = dayjs(current)
...
```

File: ui/packages/tidb-dashboard-lib/src/apps/Overview/components/LimitTimeRange.tsx [hunk 8]
```diff
@@ -67,13 +70,15 @@ export const LimitTimeRange: React.FC<LimitTimeRangeProps> = ({
     // current hour
     const hour = dayjs().hour()
     const minute = dayjs().minute()
```
Comment (baurine) [inline ui/packages/tidb-dashboard-lib/src/apps/Overview/components/LimitTimeRange.tsx:72]:
```
const today = dayjs()
const hour = today.hour()
const minute = today.minute()
```

**Review by baurine — APPROVED** [review]

**Pull request merged**